pkg/catalog/endpoint.go

```diff
@@ -36,3 +36,49 @@ func (mc *MeshCatalog) GetResolvableServiceEndpoints(svc service.MeshService) ([
 	}
 	return endpoints, nil
 }
+
+// ListAllowedEndpointsForService returns only those endpoints for a service that belong to the allowed outbound service accounts
+// for the given downstream identity
+func (mc *MeshCatalog) ListAllowedEndpointsForService(downstreamIdentity service.K8sServiceAccount, upstreamSvc service.MeshService) ([]endpoint.Endpoint, error) {
+	var outboundEndpoints []endpoint.Endpoint
+	outboundEndpoints, err := mc.ListEndpointsForService(upstreamSvc)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error looking  up endpoints for service %s with downstream identity %s", upstreamSvc, downstreamIdentity)
+		return nil, err
+	}
+	log.Trace().Msgf("Outbound endpoints for service %s with downstream identity %s: %v", upstreamSvc, downstreamIdentity, outboundEndpoints)
```

Comment:
We can remove this log if it was just added for debugging during development and isn't required.